scanpy/preprocessing/_scrublet/pipeline.py

```diff
@@ -0,0 +1,79 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Literal
+
+import numpy as np
+from scipy import sparse
+
+from .sparse_utils import sparse_multiply, sparse_var, sparse_zscore
+
+if TYPE_CHECKING:
+    from ..._utils import AnyRandom
+    from .core import Scrublet
+
+
+def mean_center(self: Scrublet) -> None:
+    gene_means = self._counts_obs_norm.mean(0)
+    self._counts_obs_norm = sparse.csc_matrix(self._counts_obs_norm - gene_means)
+    if self._counts_sim_norm is not None:
+        self._counts_sim_norm = sparse.csc_matrix(self._counts_sim_norm - gene_means)
```

Comment:
```suggestion
def mean_center(scrublet: Scrublet) -> None:
    gene_means = scrublet._counts_obs_norm.mean(0)
    scrublet._counts_obs_norm = sparse.csc_matrix(scrublet._counts_obs_norm - gene_means)
    if scrublet._counts_sim_norm is not None:
        scrublet._counts_sim_norm = sparse.csc_matrix(scrublet._counts_sim_norm - gene_means)
```
Although, I think it would make sense to make these class methods.  I mean `self` is already in the name.  And this would allow arguments to be passed into them a bit easier or things set.  Also there are loads of "private" attributes here being used, which seems like a red flag for the design.

Reply:
This seems like the sort of API we want to avoid in scanpy 2.0 to be honest.  I don't know if this is out of scope here, maybe a follow up PR is needed, but an alternative to making the `Scrublet` class all-encompassing would be a `ScrubletPipeline` class.

Reply:
I do understand the need/desire though to just make as few changes as possible to the original `scrublets` package, though.  @ivirshup should weigh in.

Reply:
Yeah, I would personally do things quite differently, but this PR is mainly about getting the tests right and replacing the parts that scanpy already has.

Reply:
@flying-sheep This was what I asked for Isaac's opinion, but I think we decided that this PR would be restricted in scope so as to not include refactoring.  I don't think we need an opinion then.

Reply:
OK, great!